<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [2]:
# Membuat dataframes dari dictionary
import pandas as pd

data = pd.DataFrame({'shio': ['Tiger','Snake','Ox','Rabbit'], 'zodiac': ['Pisces', 'Scorpio','Aries','Gemini']})
print(data)
print(type(data))

     shio   zodiac
0   Tiger   Pisces
1   Snake  Scorpio
2      Ox    Aries
3  Rabbit   Gemini
<class 'pandas.core.frame.DataFrame'>


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [ ]:
# Membaca data dan memasukannya kedalam suatu variabel


Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 

In [5]:
# code here
print("Series berisi data yang seragam (homogen) dan merupakan antara list dan dictionary, \nlist & dictionary bisa berisi berbagai tipe data. \n")
print("Dataframe ialah suatu objek dua dimensi, seperti tabel, sedangkan series ialah suatu objek satu dimensi.")

Series berisi data yang seragam (homogen) dan merupakan antara list dan dictionary, 
list & dictionary bisa berisi berbagai tipe data. 

Dataframe ialah suatu objek dua dimensi, seperti tabel, sedangkan series ialah suatu objek satu dimensi.


In [6]:
numb = pd.Series([15,3,19,98], index=['i', 'ii', 'iii', 'iv'])
print(numb)
print(type(numb))

i      15
ii      3
iii    19
iv     98
dtype: int64
<class 'pandas.core.series.Series'>


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [35]:
# code here
data=pd.read_csv('diamonds.csv')

In [36]:
data.head()

Unnamed: 0  carat      cut color clarity  depth  table  price     x     y  \
0           1   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98   
1           2   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84   
2           3   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07   
3           4   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23   
4           5   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35   

      z  
0  2.43  
1  2.31  
2  2.31  
3  2.63  
4  2.75

In [37]:
data.tail()

Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
53935  5.76  3.50  
53936  5.75  3.61  
53937  5.68  3.56  
53938  6.12  3.74  
53939  5.87  3.64

In [49]:
data.price.head()

0    326
1    326
2    327
3    334
4    335
Name: price, dtype: int64

In [48]:
data.color.tail()

53935    D
53936    D
53937    D
53938    H
53939    D
Name: color, dtype: object

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: Metode yang berhubungan dengan pengumpulan atau penyajian data berguna untuk menyampaikan informasi yang berguna dan biasanya dalam bentuk tabel, grafik, dan diagram.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [41]:
# code here
data.describe()

Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000      2.910000  
50%        5.700000      5.710000      3.530000  
75%        6.540000      6.540000      4.040000  
max       10.740000     58.900000     31.800000

Jawab: rata-rata price = 3932.799722, standar deviasi depth = 1.432621, max carat = 5.01

In [50]:
data.price.mean()

3932.799721913237

In [51]:
data.depth.std()

1.4326213188337733

In [52]:
data.carat.max()

5.01